src/legacy/core_plugins/telemetry/server/collection_manager.ts

```diff
@@ -47,10 +47,10 @@ export interface StatsCollectionConfig {
 
 export type StatsGetterConfig = UnencryptedStatsGetterConfig | EncryptedStatsGetterConfig;
 export type ClusterDetailsGetter = (config: StatsCollectionConfig) => Promise<ClusterDetails[]>;
-export type StatsGetter = (
+export type StatsGetter<T = unknown> = (
```

Comment:
we should eventually define this as a `T extends MinimumRequiredStats` so we force ourselves to ensure the minimum common payload